internal/factsengine/gatherers/sbddump.go

```diff
@@ -0,0 +1,147 @@
+package gatherers
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SBDDumpGathererName = "sbd_dump"
+)
+
+// nolint:gochecknoglobals
+var (
+	SBDDevicesLoadingError = entities.FactGatheringError{
+		Type:    "sbd-devices-loading-error",
+		Message: "error loading the configured sbd devices",
+	}
+
+	SBDDumpCommandError = entities.FactGatheringError{
+		Type:    "sbd-dump-command-error",
+		Message: "error while executing sbd dump",
+	}
+)
+
+type SBDDumpGatherer struct {
+	executor    utils.CommandExecutor
+	sbdGatherer *SBDGatherer
+}
+
+func NewDefaultSBDDumpGatherer() *SBDDumpGatherer {
+	return NewSBDDumpGatherer(utils.Executor{}, NewDefaultSBDGatherer())
+}
+
+func NewSBDDumpGatherer(executor utils.CommandExecutor, sbdGatherer *SBDGatherer) *SBDDumpGatherer {
+	return &SBDDumpGatherer{
+		executor:    executor,
+		sbdGatherer: sbdGatherer,
+	}
+}
+
+func (gatherer *SBDDumpGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s facts gathering process", SBDDumpGathererName)
+
+	configuredDevices, err := loadDevices(gatherer.sbdGatherer)
+
+	if err != nil {
+		return facts, SBDDevicesLoadingError.Wrap(err.Error())
+	}
+
+	for _, factRequest := range factsRequests {
+		var fact entities.Fact
+
+		if devicesDumps, err := getSBDDevicesDumps(gatherer.executor, configuredDevices); err == nil {
+			fact = entities.NewFactGatheredWithRequest(factRequest, &entities.FactValueMap{Value: devicesDumps})
+		} else {
+			gatheringError := SBDDumpCommandError.Wrap(err.Error())
+
+			fact = entities.NewFactGatheredWithError(factRequest, gatheringError)
+		}
+		facts = append(facts, fact)
+	}
+
+	log.Infof("Requested %s facts gathered", SBDDumpGathererName)
+	return facts, nil
+}
+
+func loadDevices(gatherer *SBDGatherer) ([]string, error) {
+	sbdDevicesRequest := []entities.FactRequest{
```

Comment:
Guess that's wiser indeed.

Initially gatherer composition sounded interesting, to avoid duplicating some the code from another gatherer, yet I realize that in this case _a little duplication is better than a little dependency_ :sweat_smile: .

Changing to `cluster.LoadSbdConfig`